### PR TITLE
updates for Ruby2.3.5 and https://github.com/SerpicoProject/Serpico/iss…

### PR DESCRIPTION
**Pull request by BuffaloWill**
This commit brings Serpico up to Ruby 2.3.5 and adds support for SSL Cipher settings (https://github.com/SerpicoProject/Serpico/issues/334) 

[edited should be 2.3.5, not 3.5]

### CODE REVIEW

File: serpico.rb [hunk 1]
```diff
@@ -5,6 +5,12 @@
 require "./server.rb"
 config_options = JSON.parse(File.read('./config.json'))
 
+# SSL Ciphers
+CIPHERS = ['ECDHE-RSA-AES128-GCM-SHA256','ECDHE-RSA-AES256-GCM-SHA384',
```
Comment (anantshri) [inline serpico.rb]: Any chance we can make it customizable variable. that ways we leave the headache of matching the ciphers as per company requirements. SSL requirements keep changing and we don't want to be the one holding back anyone.

File: serpico.rb [hunk 2]
```diff
@@ -31,6 +37,7 @@
   server_options[:SSLCertificate] = OpenSSL::X509::Certificate.new(certificate_content)
   server_options[:SSLPrivateKey] = OpenSSL::PKey::RSA.new(key_content)
   server_options[:SSLVerifyClient] = OpenSSL::SSL::VERIFY_NONE
+  server_options[:Ciphers] = CIPHERS
```
Comment (anantshri) [inline serpico.rb]: Can we also have options around selecting TLS Versions. with 1.1. and 1.0. Just being a bit too greedy am i :P

Comment (BuffaloWill) [reply serpico.rb]: Yes to both of the requests. My plan was to add it after this PR.  Any chance you tested the Ruby upgrade? Looking for feedback to make sure it doesn't break anything :smile:

Comment (anantshri) [reply serpico.rb]: unfortunately i am not in a position right now. Besides this being Q4, Personal Projects needing attention, laptop crashing, i am spread thin for time right now.

**Pull request merged**